### PR TITLE
Make ids of Markdown sub-documents prefixed with the parent item id

### LINKED ISSUE #186: Footnote appears in all following objects after the first appearance
Hi, I'm not sure this is related to pytkdocs or mkdocstrings.

**Describe the bug**
If a footnote is added to one of python objects, the footnote will appear in every object after that one in the same documentation page.  

**To Reproduce**
Here is an example to reproduce

```python
# example.py
def C():
    """This is C
    """
    pass

def B():
    """This is B[^1]

    [^1]: This is footnote
    """
    pass

def A():
    """This is A
    """
    pass
````
```
# mkdocs.yml
site_name: Footnote
theme:
  name: "material"
plugins:
  - search
  - mkdocstrings:
      default_handler: python
markdown_extensions:
  - footnotes
```
**Expected behavior**
The footnote is supposed to appear only under function B

**Screenshots**
However, footnote appears under function B and C

[
Screenshot from 2020-12-05 00-30-54
](url)

### LINKED ISSUE #193: Content tabs from different modules in the same markdown cross link to the first one always
Not sure if this is related to mkdocs, mkdocstrings, or pytkdocs.
When using content tabs with the Tabbed extension:

``` yaml
markdown_extensions:
  - pymdownx.tabbed
```

within different module docstrings, which are rendered on the same page, their IDs are regenerated and the following content tabs link to the first content tabs when clicked.

To reproduce, add a content tab block in the docstring of two modules:

``` yaml
=== "Tab1"
   some content

=== "Tab2"
   some content2

```

and then add the modules in the same markdown:

``` yaml
::: module1

::: module2

```

### PR DESCRIPTION
**Pull request by oprypin**
Fixes #193
https://github.com/pawamoy/mkdocstrings/issues/193#issuecomment-749134463

Fixes #186 
https://github.com/pawamoy/mkdocstrings/issues/186#issuecomment-743469671

### CODE REVIEW

**Comment by oprypin** [comment]
Check out this test site with both tabs and footnotes with same ids working correctly: 
https://gist.github.com/6d82074d3539851e6eaec6dc8db72dd2

This also fixes the issue I had mentioned where same headings in different docstrings would share the same id.

`#a.foo`, `#examples`, `#b.foo`, `#examples` becomes
`#a.foo`, `#a.foo--examples`, `#b.foo`, `#b.foo--examples`

**Comment by oprypin** [comment]
View test site: http://pryp.in/p/6d82074d3539851e6eaec6dc8db72dd2

File: src/mkdocstrings/handlers/base.py [hunk 1]
```diff
@@ -369,6 +375,53 @@ def teardown(self):
         self._handlers.clear()
 
 
+class _IdPrependingTreeprocessor(Treeprocessor):
+    def __init__(self, md, id_prefix: str):
+        super().__init__(md)
+        self.id_prefix = id_prefix
+
+    def run(self, root: Element):
```
Comment (pawamoy) [inline src/mkdocstrings/handlers/base.py:383]:
```suggestion
    def run(self, root: Element):  # noqa: WPS231 (not complex)
```

**Comment by pawamoy** [comment]
Once again, fantastic contribution @oprypin :rocket: Thanks!
LGTM, lets see how it behaves in the regressions workflow :slightly_smiling_face:

**Comment by oprypin** [comment]
I notice that with this change, it is now possible link to stuff like `[foo][some.identifier--subdoc-heading]`, and people are figuring it out. Should we embrace this or lock it down?

**Comment by pawamoy** [comment]
I think we should embrace it, this is a pretty nice feature. What's your opinion?

**Comment by oprypin** [comment]
Overall I support it, I'm just disgruntled by the fact that I didn't realize this effect when I developed this PR, and also the choice of the "`--`" separator wasn't with this usage in mind, although maybe there isn't anything better.

**Comment by oprypin** [comment]
For example, maybe people would've wanted to use `some.identifier#subdoc-heading` - but on the other hand, I'm already using the `#` symbol for identifiers in Crystal language, so it'd be ambiguous.

**Comment by pawamoy** [comment]
I like `--`. Headings IDs get slugified with `-` as separators, so a double `--` makes sense to me to delimit the object identifier and the subheading. A bit like Django's ORM parameters using double underscores to avoid mistaking for another attribute: `attribute__le`.

**Pull request merged**